scripts.js

```diff
@@ -0,0 +1,327 @@
+}
+
+console.log(isPalindrome('Kayak')); // true
+console.log(isPalindrome('Racecar')); // true
+console.log(isPalindrome('Was it a cat I saw')); // true
+console.log(isPalindrome('Hello!')); // false
+
+//6
+function countLetters(string) {
+  let countedLetters= {};
+  let characters = string.replace(/\s/g, '');
+  characters = characters.toLowerCase();
+
+  for (let i = 0; i < characters.length; ++i){
+    let character = characters.charAt(i)
+    if (!countedLetters[character]) {
+      countedLetters[character] = 1;
+    }
+    else {
+      countedLetters[character] += 1;
+    }
+  }
+
+  return countedLetters;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+function getYoungestPerson(peopleArray) {
+  let youngestPersonAge = peopleArray[0].age;
+
+  for (let i = 0; i < peopleArray.length; ++i) {
+    if (youngestPersonAge > peopleArray[i].age) {
+      youngestPersonAge = peopleArray[i].age;
+    }
+  }
+
+  return youngestPersonAge;
+}
+
+function getOldestPerson(peopleArray) {
+  let oldestPersonAge = 0;
+
+  for (let i = 0; i < peopleArray.length; ++i) {
+    if (oldestPersonAge < peopleArray[i].age) {
+      oldestPersonAge = peopleArray[i].age;
+    }
+  }
+
+  return oldestPersonAge;
+}
+
+function getAgeDifference(peopleArray) {
+  return getOldestPerson(peopleArray) - getYoungestPerson(peopleArray);
+}
+
+const peopleArray = [
+  {
+    name: 'Adam',
+    age: 20
+  },
+  {
+    name: 'Amanda',
+    age: 5
+  },
+  {
+    name: 'John',
+    age: 75
+  },
+  {
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getYoungestPerson(peopleArray));
+console.log(getOldestPerson(peopleArray));
+console.log(getAgeDifference(peopleArray)); // 70
+
+// 8
+function positiveSum(numbers) {
+  let numbersSum = 0;
+
+  for (let i = 0; i < numbers.length; ++i) {
+    if (numbers[i] > 0) {
+      numbersSum = numbersSum + numbers[i];
+    }
+  }
+
+  return numbersSum;
+}
+
+console.log(positiveSum([1, 5, 10]));
+console.log(positiveSum([-1, -5, -10]));
+
+// 9
+function squaredNumbersSummary(numbers) {
+  let total = 0;
+
+  for (let i = 0; i < numbers.length; ++i) {
+    total = total + numbers[i] * numbers[i];
+  }
+
+  return total;
+}
+
+console.log(squaredNumbersSummary([0, 3, 4, 5]));
+
+// 10
+function abbreviationName(name) {
+  let words = name.split(/\s+/);
+  let abbreviation = '';
+  for (let i = 0; i < words.length; ++i) {
+    abbreviation += words[i][0] + '.';
+  }
+  return abbreviation.toUpperCase().substring(0, abbreviation.length - 1);
+}
+
+console.log(abbreviationName('Kuba klin'));
+
+// 11
+function countSheep(sheep) {
+  let presenetSheep = 0;
+
+  for (let i = 0; i < sheep.length; ++i) {
+    if (sheep[i] === true) {
+      presenetSheep = presenetSheep + 1;
+    }
+  }
+
+  return presenetSheep;
+}
+
+// 12
+function digitize(numbers) {
+  let reverseArray = [];
+  let digits = ('' + numbers).split('').reverse();
```

Comment:
Why `'' + numbers`? Please explain.

Reply:
if I understand it correctly, to turn the number into the string

Reply:
That's correct. However, this is a very **implicit** way to convert a number into a string. It's difficult to read and might be confusing.
Instead, do `String(numbers)`. It's always better to be explicit if possible.

Reply:
ok, fixed